File: docs/test-beds/dart/fixture.dart
// Dart test bed
import 'helper.dart';

class Box<T> {
  T value;
  Box(this.value);
}

class Container<T> {
  final List<Box<T>> _items = [];
  void add(Box<T> item) { _items.add(item); }
  int size() => _items.length;
}

Container<String> process(Box<String> input) {
  final c = Container<String>();
  c.add(input);
  Helper.log(c);
  return c;
}

const int FOO = 42;

File: docs/test-beds/php/fixture.php
<?php
// PHP test bed
namespace TestBed;

require_once 'helper.php';

class Box {
    public $value;
    public function __construct($value) { $this->value = $value; }
}

class Container {
    private array $items = [];
    public function add(Box $item): void { $this->items[] = $item; }
    public function size(): int { return count($this->items); }
}

function process(Box $input): Container {
    $c = new Container();
    $c->add($input);
    Helper::log($c);
    return $c;
}

const FOO = 42;
